src/schema/campaign/index.js

```diff
@@ -93,6 +93,27 @@ schema.pre('save', async function setAdvertiserName() {
   }
 });
 
+schema.virtual('portalUri').get(async function getPortalUri() {
+  const advertiser = await connection.model('advertiser').findOne({ _id: this.advertiserId }, { pushId: 1 });
+  const uri = await advertiser.get('portalUri');
+  return `${uri}/campaigns/${this.pushId}`;
+});
+
+schema.virtual('vMaterialCollectUri').get(async function getVMCU() {
```

Comment:
Can we consolidate all these virtuals to a subdoc? It's getting pretty dirty at the root :)

And why the `v` prefix? Virtual?